modin/engines/base/frame/partition_manager.py

```diff
@@ -191,7 +191,7 @@ def broadcast_apply(cls, axis, apply_func, left, right):
         return np.array(
             [
                 [
-                    part.apply(
+                    part.add_to_apply_calls(
```

Comment:
Why did you replace this in the base class only? `broadcast_apply` is overridden in the child classes.

Reply:
because overridden `broadcast_apply` already uses engine specific optimized implementation. 
For example in `Ray` engine it uses remote func
https://github.com/modin-project/modin/blob/39afc07455dd0dad1807cdb1b8b1d18c4c3d8775/modin/engines/ray/pandas_on_ray/frame/partition_manager.py#L137-L157

Reply:
Okay, I see!